Tests/DependencyInjection/AbstractMongoDBExtensionTest.php

```diff
@@ -309,6 +310,20 @@ public function testXmlBundleMappingDetection(): void
         $this->assertEquals('DoctrineMongoDBBundle\Tests\DependencyInjection\Fixtures\Bundles\XmlBundle\Document', $calls[0][1][1]);
     }
 
+    public function testNewBundleStructureXmlBundleMappingDetection(): void
+    {
+        $container = $this->getContainer('NewXmlBundle');
+        $loader    = new DoctrineMongoDBExtension();
+        $config    = DoctrineMongoDBExtensionTest::buildConfiguration(
+            ['document_managers' => ['default' => ['mappings' => ['NewXmlBundle' => []]]]]
+        );
+        $loader->load($config, $container);
+
+        $calls = $container->getDefinition('doctrine_mongodb.odm.default_metadata_driver')->getMethodCalls();
+        $this->assertEquals('doctrine_mongodb.odm.default_xml_metadata_driver', (string) $calls[0][1][0]);
```

Comment:
I guess the tests are failing because this will only be true when using Symfony >= 5.4 🤔

Reply:
Hmm, I'll have to figure out what's happening differently here compared to DoctrineBundle then because the CI build is green with the same feature PR merged and I didn't add any conditionals to the tests there.

Reply:
hmm comparing with https://github.com/doctrine/DoctrineBundle/pull/1418/files, there is no `testNewBundleStructureXmlBundleMappingDetection` test there